module6/lessons/Week5/Recursion.md

```diff
@@ -0,0 +1,178 @@
+---
+layout: page
+title: Recursive Functions
+---
+
+### Warmup
+
+<section class='call-to-action' markdown='1'>
+
+Imagine you have been given the following code challenge in an interview:
+
+Build a function that takes in a string, and returns that string reversed.  EX: `input: "Megan", output: "nageM"`.  You must adhere to the following constraints:
+* You may not use _any_ LINQ methods
+* You may not use `for`, `foreach`, or `while`
+* The function must work for strings of any length\
+
+Spend 5 minutes brainstorming what this function might look like.
+
+</section>
+
+## Recursive Functions
+
+Recursion is a technique for iterating over an operation by having a function (or method) call itself repeatedly until it arrives at a result.  In other words, recursion is a way to loop without using `while` or `for`!  Let's take a look at an example.
+
+Let's say we want to countdown from a given number - the number will always be positive, but it could be * any * integer.  We could approach this in two ways: with loops, and with recursion:
+
+```c#
+public class Program
+{
+	public static void Main()
+	{
+		Console.WriteLine("Looping:");
+		LoopingCountdown(20);
+		Console.WriteLine("Recursive:");
+		RecursiveCountdown(20);
+	}
+	
+	static void LoopingCountdown(int num)
+	{
+		while (num > -1)
+		{
+			Console.WriteLine(num);
+			num--;
+		}
+	}
+	
+	static void RecursiveCountdown(int num)
+	{
+		if (num == )
+		{
+			Console.WriteLine(num);
+		}
+		else
+		{
+			Console.WriteLine(num);
+			RecursiveCountdown(num - 1);
+		}
+	}
+}
+```
+
+This is a very simple example of recursion, but it does give us a chance to identify the key features of a recursive function.  Every recursive function must have these two pieces:
```

Comment:
I'm glad that you have these two parts written out! I find always coming back to these two things is helpful when I've taught recursion in the past.